app/services/decision-report-generation.js

```diff
@@ -1,31 +1,39 @@
 import Service, { inject as service } from '@ember/service';
 import { task } from 'ember-concurrency';
-import { deleteFile } from 'frontend-kaleidos/utils/document-delete-helpers';
 
 export default class DecisionReportGeneration extends Service {
   @service toaster;
   @service store;
   @service intl;
 
-  generateReplacementReport = task(async (report) => {
-    const fileMeta = await this.exportPdf.perform(report);
-    await this.replaceFile(report, fileMeta.id);
-  });
-
-  exportPdf = task(async (report) => {
-    const resp = await fetch(`/generate-decision-report/${report.id}`);
-    if (!resp.ok) {
+  generateReplacementReports = task(async (reports) => {
+    const generatingPDFToast = this.toaster.loading(
+      this.intl.t('decision-report-generation--toast-generating--message'),
+      this.intl.t('decision-report-generation--toast-generating--title'),
+      {
+        timeOut: 3 * 60 * 1000,
+      }
+    ); 
+    const reportIDs = reports.map((report) => { return report.id });
+    const data = {
+      reportIDs: reportIDs
+    };
+    const url = `/generate-decision-report/generate`;
+    const resp = await fetch(url, {
```

Comment:
**issue**: this call might get timed out by the browser

Since you're now passing in multiple reports, this call might take a while (e.g. for very large agendas, with 100s of reports to re-generate, this might take multiple minutes). Browsers time out calls that take longer than a minute and that's a setting that we can't change in Kaleidos (it's per-user and set locally). The simplest solution is to implement the job model, like we do for report generation, and let the backend service create a job that we can track in the frontend. This way the initial call should respond quite quickly (it should just create the job that we track and trigger it) and then we can poll the job until completion.

Some context/inspiration can be found in the code to generate reports:
- https://github.com/kanselarij-vlaanderen/frontend-kaleidos/blob/development/app/controllers/publications/overview/reports.js#L37-L67
- https://github.com/kanselarij-vlaanderen/publication-report-service

As well as the code to generate file bundles:
- https://github.com/kanselarij-vlaanderen/frontend-kaleidos/blob/development/app/components/agenda/agenda-header/agenda-actions.js#L264-L323
- https://github.com/kanselarij-vlaanderen/file-bundling-job-creation-service/blob/master/app.js

(I wouldn't bother with creating a new service to create job, rather add an endpoint to create the job to the existing decision-report-generation-service).